### PR TITLE
Support verify custom root ca

### PR DESCRIPTION
**Pull request by jhsiaosuse**
There might be a situation the syslog server owns custom Root CA and this CA don't wanna put into system certificate chain but read from a file. Also, the syslog client doesn't want to bypass certificate checks.

Therefore, support a new config field `root_ca` which specifies the location of the Root CA certificate. The syslog client plugin loads the Root CA certificate *only when* the user configures `root_ca` path and the `insecure_skip_verify` is `false` which means no to bypass certificate checks.

### CODE REVIEW

File: pkg/syslog/out.go [hunk 1]
```diff
@@ -241,18 +248,39 @@ func (s *Sink) MessagesDropped() int64 {
...
-			dialer := net.Dialer{
-				Timeout: out.dialTimeout,
+			var (
+				conn  net.Conn // conn needs to be of type net.Conn, not *tls.Conn
+				roots *x509.CertPool
+				pem   []byte
+				err   error
+			)
+
+			// load certificate when need to do certificate check and certificate path not empty.
```
Comment (wfernandes) [inline pkg/syslog/out.go]: This comment is unnecessary because the code is very clear as to what is happening.

**Comment by jhsiaosuse** [comment]
> @jhsiaosuse Thanks for the PR! I just had a couple of review points.
> 
> * I would like to see a test for the cases around this.
> * I would also like to better understand your use case for this plugin. That is, are y'all using this plugin independently and hand rolling the fluent-bit daemon set/config or is this being used in conjunction with [knative/observability](https://github.com/knative/observability)?
> 
> We are available to discuss more in slack: https://pivotal.slack.com/messages/CAYLN0QMB

@wfernandes,
- The test case added, please check. I put the certificate and key to the `testdata` folder (in Golang put the test data to the this folder).
- We use fluent-bit as a daemonset with its config map. Since we don't want to add the custom root CA to the trusted certificate chain in Dockerfile but provides the custom root CA by volume mounted.

**Comment by jhsiaosuse** [comment]
@wfernandes ping!

**Comment by wfernandes** [comment]
/lgtm

@jhsiaosuse Thanks for writing the test! I downloaded your fork:branch and ran the tests and it looks like everything passed.
Apologies for not responding earlier, I'm out traveling to KubeCon EU. 
I'll let @Benjamintf1 @hev take it from here when they get in the office on Monday.

**Comment by jhsiaosuse** [comment]
@wfernandes any updates?

**Comment by Benjamintf1** [comment]
@jhsiaosuse HI. There's a bunch of changes from a large scale refactor we've been working on for a long time that we've finally gotten in. If you want this in, you should change this pr to account for the updates.

Specifically, the configuration for sinks includes far less json and now is setup for each sink being a separate output plugin.

**Comment by Benjamintf1** [comment]
looks good to me. Verified it as well.

**Pull request merged**